Support for target msp430-none-elf

I am currently working on a project on an MSP430 and embassy-executor appears to require an implementation of `atomic-polyfill` for this target. I can cook up a patched version on a fork and submit a PR. Creating this issue to track.

Notably, it is the lack of `AtomicBool` that is the issue:

```rust
error[E0432]: unresolved import `atomic_polyfill::AtomicBool`
 --> /Users/vax/.cargo/registry/src/index.crates.io-6f17d22bba15001f/static_cell-1.0.0/src/lib.rs:7:23
  |
7 | use atomic_polyfill::{AtomicBool, Ordering};
  |                       ^^^^^^^^^^ no `AtomicBool` in the root
  ```

--- a/build.rs
+++ b/build.rs
@@ -26,6 +26,7 @@ fn main() {
 
     let patterns = [
         ("avr-*", (Polyfill, Polyfill)),
+        ("msp430-none-elf", (Polyfill, Polyfill)),
         ("riscv32imac-*", (Native, Polyfill)),
         ("riscv32gc-*", (Native, Polyfill)),
         ("riscv32imc-*-espidf", (Native, Native)),

--- a/build.rs
+++ b/build.rs
@@ -25,11 +25,11 @@ fn main() {
     use PolyfillLevel::*;
 
     let patterns = [
+        ("avr-*", (Polyfill, Polyfill)),
         ("riscv32imac-*", (Native, Polyfill)),
         ("riscv32gc-*", (Native, Polyfill)),
         ("riscv32imc-*-espidf", (Native, Native)),
         ("riscv32*", (Polyfill, Polyfill)),
-        ("avr-*", (Polyfill, Polyfill)),
         ("thumbv4t-*", (Polyfill, Polyfill)),
         ("thumbv6m-*", (Polyfill, Polyfill)),
         ("thumbv7m-*", (Native, Polyfill)),

--- a/src/polyfill.rs
+++ b/src/polyfill.rs
@@ -231,11 +231,11 @@ impl AtomicBool {
     }
 
     pub fn load(&self, _order: Ordering) -> bool {
-        return critical_section::with(|_| unsafe { *self.inner.get() });
+        critical_section::with(|_| unsafe { *self.inner.get() })
     }
 
     pub fn store(&self, val: bool, _order: Ordering) {
-        return critical_section::with(|_| unsafe { *self.inner.get() = val });
+        critical_section::with(|_| unsafe { *self.inner.get() = val });
     }
 
     pub fn swap(&self, val: bool, order: Ordering) -> bool {
@@ -390,11 +390,11 @@ impl<T> AtomicPtr<T> {
     }
 
     pub fn load(&self, _order: Ordering) -> *mut T {
-        return critical_section::with(|_| unsafe { *self.inner.get() });
+        critical_section::with(|_| unsafe { *self.inner.get() })
     }
 
     pub fn store(&self, val: *mut T, _order: Ordering) {
-        return critical_section::with(|_| unsafe { *self.inner.get() = val });
+        critical_section::with(|_| unsafe { *self.inner.get() = val });
     }
 
     pub fn swap(&self, val: *mut T, order: Ordering) -> *mut T {

